Fakebook.Profile/Fakebook.Profile.Domain/DomainProfile.cs

```diff
@@ -0,0 +1,217 @@
+﻿using System;
+using System.Text.RegularExpressions;
+
+using Fakebook.Profile.Domain.Utility;
+
+namespace Fakebook.Profile.Domain
+{
+    /// <summary>
+    /// A profile for the domain logic and Api to use.
+    /// </summary>
+    public class DomainProfile
+    {
+        //[anything]@[anything].[anything]
+        private string _email;
+        /// <summary>
+        /// The user's email.
+        /// </summary>
+        /// <remarks>
+        /// Cannot be null, and must match the Regex for email characters.
+        /// </remarks>
+        public string Email
+        {
+            get => _email;
+            set
+            {
+                if (string.IsNullOrWhiteSpace(value))
+                {
+                    throw new ArgumentNullException($"Invalid Email, {value}", nameof(value));
+                }
+
+                Regex emailRegex = new Regex(RegularExpressions.EmailCharacters);
+                if (!emailRegex.IsMatch(value))
+                    throw new ArgumentException("The email doesn't max the regex", nameof(value));
+                _email = value;
+            }
+        }
+
+        /// <summary>
+        /// A url that is expected to be an image. Technically though, no checks are done to enforce that.
+        /// </summary>
+        public Uri ProfilePictureUrl { get; set; }
+
+        /// <summary>
+        /// Shortcut to get a user's full name. 
+        /// </summary>
+        public string Name => $"{FirstName} {LastName}";
+
+        private string _firstName;
+        /// <summary>
+        /// The user's first name.
+        /// </summary>
+        /// <remarks>
+        /// A-Z, ', ., - are the only valid characters for first and last name.
```

Comment:
far too stringent. there arent many valid assumptions you can make here let alone useful ones.